tests/deepsparse/transformers/pipelines/helpers.py

```diff
@@ -82,3 +87,53 @@ def _create_tokenizer(model_name):
             tokenizer.pad_token = tokenizer.eos_token
 
         return tokenizer
+
+
+def parse_params(config_path: str) -> Tuple[Optional[Dict], Optional[str]]:
+    # parses the config file provided
+    assert os.path.isfile(config_path), f"config_path {config_path} is not a file"
+    # reads the yaml file
+    with open(config_path, "r") as f:
+        config = yaml.safe_load(f)
+
+    cadence = os.environ.get("CADENCE", "commit")
+    expected_cadence = config["cadence"]
+
+    if not isinstance(expected_cadence, list):
+        expected_cadence = [expected_cadence]
+    if cadence in expected_cadence:
+        return config, None
+    return None, "Skipping test for cadence: {}".format(config["cadence"])
+
+
+def validate_cache_management_type(
```

Comment:
I would change this code to return True or False, not the value of internal_kv_cache. 
depending on the output, I would then skip pytest or continue the test and use internal_kv_cache.

Also where is this used?

Reply:
it is used inside the `test_text_generation.py`. The cool thing is that once we hit `skipif`, we automatically skip the whole testing scenario. Else, we return True or False as you are suggesting.